src/Framework/Microsoft.Build.Framework.csproj

```diff
@@ -1,14 +1,27 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(LibraryTargetFrameworks)</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GenerateReferenceAssemblySource>true</GenerateReferenceAssemblySource>
     <CreateTlb>true</CreateTlb>
     <IsPackable>true</IsPackable>
     <PackageDescription>This package contains the $(MSBuildProjectName) assembly which is a common assembly used by other MSBuild assemblies.</PackageDescription>
     <IncludeSatelliteOutputInPack>false</IncludeSatelliteOutputInPack>
     <ApplyNgenOptimization Condition="'$(TargetFramework)' == '$(FullFrameworkTFM)'">partial</ApplyNgenOptimization>
+
+    <!--
+      Ignore warnings for nullable annotations in code that doesn't use '#nullable'. Microsoft.CodeAnalysis.Collections
+      source assumes nullable reference types is enabled, so this warning is reported but cannot be fixed in code from
+      an imported package. This suppression should be removed if/when the project is migrated to enable nullable
+      reference types.
+    -->
+    <NoWarn>$(NoWarn),CS8632</NoWarn>
```

Comment:
📝 This is certainly not the way I'd hope this would go. Once https://github.com/dotnet/roslyn/issues/52893 is implemented we can quickly deal with this.